var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -326,12 +327,9 @@ def bootstrap_args(self):
         if not sys.platform == "win32":
             args.append("--prefix={0}".format(self.prefix))
 
-            jobs = spack.build_environment.get_effective_jobs(
-                make_jobs,
-                parallel=self.parallel,
-                supports_jobserver=self.generator.supports_jobserver,
-            )
-            if jobs is not None:
+            jobs = 1 if env_flag(spack.build_environment.SPACK_NO_PARALLEL_MAKE) else make_jobs
```

Comment:
Maybe easier if `SPACK_NO_PARALLEL_MAKE` implies `not should_support_jobserver`?

Reply:
I don't see where it simplifies: we would still need to set `jobs` to `1` if `SPACK_NO_PARALLEL_MAKE` ?

I don't know the original intent behind `SPACK_NO_PARALLEL_MAKE`, but it would simplify if we make it imply `make_jobs = 1` globally.

To me `SPACK_NO_PARALLEL_MAKE` looks like it means "disable parallel only for `make` commands", but even the current implementation/usage seems a bit broader (and [comment](https://github.com/spack/spack/blob/fce95e2efb76da3c6ee549f2a9b4e257fb42ae50/lib/spack/spack/build_environment.py#L87)).

Reply:
Here is a version that changes `SPACK_NO_PARALLEL_MAKE` behavior to disable all parallel builds (`make_jobs = 1`), and with subsequent simplifications in recipes: [compared to this MR](https://github.com/Jordan474/spack/compare/simpler-jobserver-support...simpler-jobserver-support-2), and [compared to develop](https://github.com/spack/spack/compare/develop...Jordan474:spack:simpler-jobserver-support-2).

Does it look better to you? should I merge that in this PR?

Reply:
Here is the version with an even simpler API: https://github.com/spack/spack/pull/35644 .